2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,382 @@
+
+        assertThat(elementValue).isEqualTo(228);
+    }
 
-        assertFalse(integerQueue.isEmpty());
+    @Test
+    @Order(12)
+    void addElementIntoNotEmptyQueue() {
+        addIntElementToQueue(12);
+
+        int sizeBeforeAdd = getInternalSize();
+        boolean isEmptyBeforeAdd = isEmptyQueue();
+
+        integerQueue.add(111);
+        int sizeAfterAdd = getInternalSize();
+        boolean isEmptyAfterAdd = isEmptyQueue();
+        Integer firstValue = (Integer) pollElementFromQueue();
+        Integer secondValue = (Integer) pollElementFromQueue();
+
+        assertThat(sizeBeforeAdd).isEqualTo(1);
+        assertThat(sizeAfterAdd).isEqualTo(2);
+
+        assertThat(isEmptyBeforeAdd).isEqualTo(false);
+        assertThat(isEmptyAfterAdd).isEqualTo(false);
+
+        assertThat(firstValue).isEqualTo(12);
+        assertThat(secondValue).isEqualTo(111);
+    }
+
+    private Class<?> getInnerStaticNodeClass() {
+        return Arrays.stream(integerQueue.getClass().getDeclaredClasses())
+                .filter(aClass -> Modifier.isStatic(aClass.getModifiers()))
+                .findAny()
+                .orElseThrow();
+    }
+
+    private boolean hasField(Class<?> classToSearch, Predicate targetField) {
+        return Arrays.stream(classToSearch.getDeclaredFields())
+                .anyMatch(targetField);
+    }
+
+    @SneakyThrows
+    private int getInternalSize() {
+        return (int) getAccessibleFieldByPredicate(this.integerQueue, SIZE_FIELD)
+                .get(this.integerQueue);
+    }
+
+
+    @SneakyThrows
+    private Object pollElementFromQueue() {
+        Object element;
+        Object nextElement;
+        Object tail;
+        Object head = getAccessibleFieldByPredicate(this.integerQueue,
+                HEAD_FIELD)
+                .get(this.integerQueue);
+
+        Integer size = (Integer) getAccessibleFieldByPredicate(this.integerQueue,
+                SIZE_FIELD)
+                .get(this.integerQueue);
+
+        if (head != null) {
+            element = getAccessibleFieldByPredicate(head, ELEMENT_FIELD)
+                    .get(head);
+
+            nextElement = getAccessibleFieldByPredicate(head, NODE_FIELD)
+                    .get(head);
+
+            head = nextElement;
+            setHead(head);
+
+            if (head == null) {
+                tail = null;
+                setTail(tail);
+            }
+
+            if (size != null) {
+                int tmpInt = size;
+                tmpInt--;
+                setInternalSize(tmpInt);
+            }
+
+            return element;
+        } else {
+            return null;
+        }
+    }
+
+    @SneakyThrows
+    private void addIntElementToQueue(int value) {
+        Class<?> innerClass = getInnerStaticNodeClass();
+        Constructor constructor = innerClass.getDeclaredConstructor(Object.class);
+        constructor.setAccessible(true);
+        Object nodeObj = constructor.newInstance(value);
```

Comment:
Imagine implementation of Node class without constructor with argument. The tests won't work in this case. You can find right solution in LinkedList tests.